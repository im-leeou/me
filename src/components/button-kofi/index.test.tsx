import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { ButtonKofi } from '.';

describe('components button kofi', () => {
  it('renders correctly', () => {
    const result = render(<ButtonKofi>Click me</ButtonKofi>);

    expect(result).toMatchSnapshot();
  });

  it('renders as full width', () => {
    const result = render(<ButtonKofi fullW>Click me</ButtonKofi>);

    expect(result).toMatchSnapshot();
  });
});
