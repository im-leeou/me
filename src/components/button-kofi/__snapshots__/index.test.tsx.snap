// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components button kofi > renders as full width 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        <div
          class="relative mr-2 flex size-5 md:size-6"
        >
          <img
            alt="im-kofi-24"
            class="m-auto !h-auto"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        Kofi me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
      >
        <div
          class="relative mr-2 flex size-5 md:size-6"
        >
          <img
            alt="im-kofi-24"
            class="m-auto !h-auto"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        Kofi me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
    >
      <div
        class="relative mr-2 flex size-5 md:size-6"
      >
        <img
          alt="im-kofi-24"
          class="m-auto !h-auto"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      Kofi me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components button kofi > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        <div
          class="relative mr-2 flex size-5 md:size-6"
        >
          <img
            alt="im-kofi-24"
            class="m-auto !h-auto"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        Kofi me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
    >
      <div
        class="relative mr-2 flex size-5 md:size-6"
      >
        <img
          alt="im-kofi-24"
          class="m-auto !h-auto"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fim-kofi-24.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      Kofi me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
